### PR TITLE
CONSOLE-3220: Add console capability to all manifests

### PR DESCRIPTION
**Pull request by raspbeep**
https://issues.redhat.com/browse/CONSOLE-3220

Signed-off-by: Pavel Kratochvil <pakratoc@redhat.com>

### CODE REVIEW

**Review by jhadvig — COMMENTED** [review]
@raspbeep thanks for the PR! We also need to add the annotation manifests in the `bindata/`  and `quickstarts/` folders.

**Review by spadgett — COMMENTED** [review]
It looks like we're leaving the console CRDs (ConsoleLink, ConsolePlugin, etc) even if console is removed. Is that right?

We plan to write a conversion webhook for the ConsolePlugin CRD in the console operator, so we won't be able to have the CRD without the operator. We need to decide if

* We want to remove the console CRDs or
* We leave the both the console CRDs and operator (for the webhook) even if console is removed or
* We create a separate pod just for the webhook.

@jhadvig @bparees Opinion? I'm inclined to say we remove the CRDs, but any operators that relies on them would need to tolerate it when they're missing.

**Comment by bparees** [comment]
> @jhadvig @bparees Opinion? I'm inclined to say we remove the CRDs, but any operators that relies on them would need to tolerate it when they're missing.

i'm also inclined to remove them, i'm not sure what background/motivation lead to the decision to always enable them.  Presumably it was to avoid breaking consumers of those apis, as @spadgett said, but we should be moving them towards tolerating the absence of these apis, as we make the console in general an optional component.

**Comment by jhadvig** [comment]
> It looks like we're leaving the console CRDs (ConsoleLink, ConsolePlugin, etc) even if console is removed. Is that right?

@spadgett @bparees So if the console-operator wont part of the cluster, CVO wont be pulling all it's manifests (together with Console's CRDs) from the image and creating them. For that reason the CRD's should not be created on the cluster if console-operator is disabled. Or is my understanding wrong?

**Comment by spadgett** [comment]
@jhadvig My understanding is the CVO will still install any manifests that don't have the capability annotation. Since we copy the CRD manifests from openshift/api in our Dockerfile, those would still get installed since they don't have the annotation. (Presumably there is a way to add the annotation to the generated CRD from the api repo?)

@bparees Can you confirm?

**Comment by bparees** [comment]
> My understanding is the CVO will still install any manifests that don't have the capability annotation. Since we copy the CRD manifests from openshift/api in our Dockerfile, those would still get installed since they don't have the annotation.

correct.  What matters is the annotations on the resource yamls that go into the payload.


> (Presumably there is a way to add the annotation to the generated CRD from the api repo?)

i think you can just add it manually and generation doesn't touch it.

**Comment by jhadvig** [comment]
@raspbeep could you please update https://github.com/openshift/api/pull/1232 and at the annotation to our CRDs ?

**Comment by raspbeep** [comment]
Thanks for the reviews. Added the annotations to CRDs in https://github.com/openshift/api/pull/1232.

File: bindata/configmaps/console-configmap.yaml
Comment (spadgett) [inline bindata/configmaps/console-configmap.yaml:8]: I don't believe this needs to be on anything the operator itself creates (in other words anything in `bindata`). Just the YAML files in `manifests`.

**Comment by jhadvig** [comment]
@spadgett you are right. Since operator is working with bindata we dont need to add the annotation to them. On the other hand we should probably add it to the `quickstarts/` CRs.

**Comment by spadgett** [comment]
> @spadgett you are right. Since operator is working with bindata we dont need to add the annotation to them. On the other hand we should probably add it to the `quickstarts/` CRs.

Right, the quick start CRs will need it.

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by jhadvig** [comment]
/hold

QE Approver:
 /assign @yanpzhan 
Docs Approver:
 /assign @opayne1
PX Approver:
/assign @RickJWagner

**Comment by yanpzhan** [comment]
Since we could not test console capability feature against this pr directly, will test the feature after merge.

**Comment by wking** [comment]
[e2e-gcp][1]:

```
level=error msg=waiting for openshift-console URL: context deadline exceeded
```

So I expect you'll want to revive openshift/installer#5336 or do something similar to allow installs that exclude the console component.

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_console-operator/665/pull-ci-openshift-console-operator-master-e2e-gcp/1554086279157649408#1:build-log.txt%3A69-70

File: manifests/0000_90_console-operator_01_prometheusrbac.yaml
Comment (spadgett) [inline manifests/0000_90_console-operator_01_prometheusrbac.yaml:11]: Should these be lowercase `console`?

Comment (bparees) [reply manifests/0000_90_console-operator_01_prometheusrbac.yaml:11]: no, we're pivoting to PascalCase for the enum values

**Comment by wking** [comment]
Hmm, I was checking in back here to report openshift/installer#5336 landing to unblock your CI.  But weirdly, [this run from the 23rd][1] has:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_console-operator/665/pull-ci-openshift-console-operator-master-e2e-gcp/1562005782285783040/artifacts/e2e-gcp/ipi-install-install/artifacts/.openshift_install.log | grep -A5 'openshift-console route'
time="2022-08-23T10:11:59Z" level=info msg="Waiting up to 10m0s (until 10:21AM) for the openshift-console route to be created..."
time="2022-08-23T10:11:59Z" level=debug msg="Route found in openshift-console namespace: console"
time="2022-08-23T10:11:59Z" level=debug msg="OpenShift console route is admitted"
time="2022-08-23T10:11:59Z" level=info msg="Install complete!"
time="2022-08-23T10:11:59Z" level=info msg="To access the cluster as the system:admin user when using 'oc', run\n    export KUBECONFIG=/tmp/installer/auth/kubeconfig"
time="2022-08-23T10:11:59Z" level=info msg="Access the OpenShift web-console here: https://console-openshift-console.apps.ci-op-zph4n5kx-2f88d.XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
```

Is the plan to leave that route in place, even when the console is removed?  Ah, it's because openshift/cluster-version-operator#801 landed:

```console
$ curl -s https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_console-operator/665/pull-ci-openshift-console-operator-master-e2e-gcp/1562005782285783040/artifacts/e2e-gcp/gather-extra/artifacts/clusterversion.json | jq -c '.items[].status.capabilities.enabledCapabilities'
["Console","Insights","Storage","baremetal","marketplace","openshift-samples"]
```

and that took CI on this PR from "will things work with our removal annotations?" to "meaningless" :joy:.  I guess we'll see post-merge ;)

[1]: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_console-operator/665/pull-ci-openshift-console-operator-master-e2e-gcp/1562005782285783040

**Comment by jhadvig** [comment]
/retest

```
ERRO[2022-08-29T23:04:00Z] Some steps failed:                           
ERRO[2022-08-29T23:04:00Z] 
  * could not run steps: step e2e-aws-console failed: "e2e-aws-console" test steps failed: "e2e-aws-console" pod "e2e-aws-console-test" failed: the pod ci-op-h8hckbiq/e2e-aws-console-test failed after 2h1m8s (failed containers: test): ContainerFailed one or more containers exited
Container test exited with code 1, reason Error
---
ng Container image with extrenal registry on Add page: KN-05-TC05 (example #1) (376617ms)
    ✓ knative service menu options: KN-02-TC01 (9737ms)
    ✓ side bar details of knative Service: KN-06-TC01 (11087ms)
    ✓ Edit labels modal details: KN-02-TC02 (10570ms)
    ✓ Edit Annotation modal details: KN-02-TC17 (9932ms)
      Verifying service kn-service is present in application openshift-app
    ✓ Update the service to new application group: KN-02-TC08 (18182ms)
    ✓ Context menu for knative Revision: KN-01-TC01 (8584ms)
    ✓ side bar details of knative Revision: KN-06-TC02 (10495ms)
    ✓ Delete Revision not possible for the service which contains one revision: KN-01-TC12 (10485ms)
      user is able to see revisions in knative service : kn-service of topology side pane
    ✓ Create Revision for the existing knative Service (17777ms)
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 2h0m0s timeout","severity":"error","time":"2022-08-29T22:50:01Z"}
      Displaying following error: "admission webhook "validation.webhook.serving.knative.dev" denied the request: validation failed: Traffic targets sum to 150, want 100: spec.traffic", so closing this form or modal
    ✓ Set traffic distribution greater than 100% for the Revisions of the knative Service: KN-02-TC10 (12036ms)
{"component":"entrypoint","file":"k8s.io/test-infra/prow/entrypoint/run.go:255","func":"k8s.io/test-infra/prow/entrypoint.gracefullyTerminate","level":"error","msg":"Process did not exit before 15s grace period","severity":"error","time":"2022-08-29T22:50:16Z"}
{"component":"entrypoint","error":"process timed out","file":"k8s.io/test-infra/prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2022-08-29T22:50:16Z"}
error: failed to execute wrapped command: exit status 127
```

**Comment by raspbeep** [comment]
Thank you, @invincibleJai 🙏

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

**Comment by jhadvig** [comment]
/retest
```
    unmanaged_test.go:12: waiting for setup to reach settled state...
    unmanaged_test.go:13: changing console operator state to 'Unmanaged'...
    util.go:28: patching Data on the console ConfigMap
    util.go:35: polling for patched Data on the console ConfigMap
    unmanaged_test.go:41: error: timed out waiting for the condition
    unmanaged_test.go:18: waiting for cleanup to reach settled state...
--- FAIL: TestEditUnmanagedConfigMap (15.90s)
```
looks like a flake

**Pull request merged**
